### ISSUE TITLE
Cannot build rules_haskell using the protobuf 3.7.x line.

### ISSUE THREAD

**Issue by NinjaTrappeur** [tweag/rules_haskell#619]
When building rules_haskell on Nixos with protobuff `1069565a68654fa016cb454c87eca8372b3b8d3f`, the build end up failing with:

```
$ bazel build //... -s --verbose_failures --sandbox_debug
INFO: Invocation ID: a0b0be2f-659d-445f-8986-13797fd98662
ERROR: .cache/bazel/_bazel_minoulefou/5d77ce41c91424d253ef2f271854858b/external/io_tweag_rules_nixpkgs/nixpkgs/nixpkgs.bzl:197:9: no such target '@nixpkgs//:nixpkgs': target 'nixpkgs' not declared in package ''; however, a source file of this name exists.  (Perhaps add 'exports_files(["nixpkgs"])' to /BUILD?) defined by .cache/bazel/_bazel_minoulefou/5d77ce41c91424d253ef2f271854858b/external/nixpkgs/BUILD and referenced by '//external:zlib'
ERROR: Analysis of target '//docs:api_html' failed; build aborted: Analysis failed
```

**Comment by mboes** [comment]
I doubt this error is protobuf-specific. And I doubt this is a rules_haskell issue (likely a rules_nixpkgs or protobuf one). Do you have a more complete repro? It's possible that with newer Bazel's we need the `exports_files` line, rather than the empty `BUILD` file we create here: https://github.com/tweag/rules_nixpkgs/blob/master/nixpkgs/nixpkgs.bzl#L7.

**Comment by NinjaTrappeur** [comment]
Indeed, seems like rules_nixpkgs build env is still using bazel 0.15.

I'm going to look for the complete repro, fix this and the other possible bazel-based breakages.

**Comment by NinjaTrappeur** [comment]
I'm actually not sure the issue is linked to `rules_nixpkgs`.

After generating a proper `BUILD` file, I end up with the error

```
ERROR:.cache/bazel/_bazel_minoulefou/5d77ce41c91424d253ef2f271854858b/external/io_tweag_rules_nixpkgs/nixpkgs/nixpkgs.bzl:198:9: in repository attribute of _nixpkgs_package rule //external:zlib: source file '@nixpkgs//:nixpkgs' is misplaced here (expected no files)
ERROR: Analysis of target '//tests/haskell_proto_library:address_proto' failed; build aborted: Analysis of target '//external:zlib' failed; build aborted
```

This error does not appears if we do not use the protobuf 3.7.x line. I have not been able to reproduce this with a lib-dependent test-rule in nixpkgs alone.

The rule generating the error, `//external:zlib` comes from protobuf: https://github.com/protocolbuffers/protobuf/blob/master/BUILD#L22

I don't get how `//external:zlib` ends up being in a `nixpkgs_package` rule. Is `//external` a special target? I did not find anything  on Bazel's doc and am confused as hell.

**Comment by Profpatsch** [comment]
`//external` is mentioned here: https://docs.bazel.build/versions/master/query.html#external-repos

**Comment by Profpatsch** [comment]
Is this still an issue?

**Comment by NinjaTrappeur** [comment]
AFAIK, yes. I did not fix this nor seen anything fixing it merged into `rules_haskell`.

I can't dive on this ATM, but I feel like this may be an integration problem between `rules_nixpkgs` and the latest proto rules.

**Comment by mboes** [comment]
The issue was likely due to the fact that protobuf was stealthily bringing into scope a package called `@zlib`, which clashed with  the `@zlib` where were bringing in scope as a `nixpkgs_package`. This was fixed in d5ef2808f32aacaffce9a783a0aaff59ab59d702. If this is still a problem, reopen the issue.
